klippa/src/repack.rs

```diff
@@ -64,10 +77,147 @@ pub(crate) fn resolve_graph_overflows(
     if overflows.is_empty() {
         Ok(())
     } else {
+        if (tag == Gsub::TAG || tag == Gpos::TAG) && !always_recalculate_extensions {
+            return resolve_graph_overflows(graph, tag, max_round, true);
+        }
         Err(RepackErrorFlags::REPACK_ERROR_NO_RESOLUTION)
     }
 }
 
+fn promote_extensions_if_needed(
+    graph: &mut Graph,
+    lookup_list_idx: ObjIdx,
+    lookups: &[ObjIdx],
+    table_tag: Tag,
+) -> Result<(), RepackErrorFlags> {
+    struct LookupSize {
+        obj_idx: ObjIdx,
+        lookup_size: usize,
+        subgraph_size: usize,
+        subtable_count: usize,
+        lookup_type: u16,
+        is_ext: bool,
+    }
+
+    impl LookupSize {
+        fn cmp(&self, other: &LookupSize) -> std::cmp::Ordering {
+            let bytes_per_subtable_a = self.subtable_count as f64 / self.subgraph_size as f64;
+            let bytes_per_subtable_b = other.subtable_count as f64 / other.subgraph_size as f64;
+            if bytes_per_subtable_b < bytes_per_subtable_a {
+                Ordering::Less
+            } else if bytes_per_subtable_b > bytes_per_subtable_a {
+                Ordering::Greater
+            } else {
+                Ordering::Equal
```

Comment:
be aware that this will return `Equal` in unexpected cases, such as if either side is a NaN.